query-engine/schema/src/build/enum_types.rs

```diff
@@ -111,7 +111,7 @@ pub fn itx_isolation_levels(ctx: &'_ QuerySchema) -> Option<EnumType> {
 }
 
 pub(crate) fn relation_load_strategy(ctx: &QuerySchema) -> Option<EnumType> {
-    if !ctx.has_feature(psl::PreviewFeature::RelationJoins) {
+    if !ctx.can_resolve_relation_with_joins() {
```

Comment:
This is technically a breaking change, but it's fine for the preview feature. We used to not include the relation load args _only_ if the preview feature wasn't set. The relation load args need to be excluded when the connector doesn't have `ConnectorCapability::RelationJoins` or `ConnectorCapability::CorrelatedSubqueries` too.